src/code.rs

```diff
@@ -470,10 +507,56 @@ impl {struct_name} {{
     ));
 
     buffer.push_str(
-        r##"
-}"##,
+        r#"
+}"#,
     );
 
+    // generate filter struct for filter() helper function
+    {
+        let filter_fields = table
+            .columns
+            .iter()
+            .map(|column| {
+                let column_name = column.name.to_string();
+                format!(
+                    "pub {column_name}: Option<{column_type}>,",
+                    column_name = column_name,
+                    column_type = if column.is_nullable {
+                        format!("Option<{}>", column.ty)
+                    } else {
+                        column.ty.clone()
+                    }
```

Comment:
i am pretty sure this will lead to errors (because int conversion is forgotten), instead use the helper type by first converting the column to a `StructField` like `StructField::from(column)`

Reply:
I thought about leveraging StructField, but I remember there were some things I bumped into that made me change my mind and took me towards this path. Let's merge this in and discuss any errors as they come up.

Things that StructField deal with (like "str/String/Cow", etc.) don't apply to this filter API

Reply:
let's merge this in for now and deal with any issues that come up

Reply:
see #65, #67 which i cleaned-up with #85

this would be required because `column` in this case is a `ParsedColumnMacro`, which still means it is "raw", which would look like this:
```txt
ParsedColumnMacro {
    ty: "i32",
    name: Ident(
        unsigned_nullable,
    ),
    is_nullable: true,
    is_unsigned: true,
}
```
so you would need to convert the `ty` `i32` to `u32` and also make it wrapped in `Options`, and `StructField` was added in #85 exactly to deduplicate this code to not have issues like this anymore